tests/flow/test_constraint.py

```diff
@@ -998,7 +998,7 @@ def __init__(self):
         self.monitor = []
         self.g = self.db.select_graph(GRAPH_ID)
 
-        self.monitor_thread = threading.Thread(target=self.monitor_thread)
+        self.monitor_thread = threading.Thread(target=self.monitor_thread, daemon=True)
```

Comment:
See if there's a way to signal the `monitor_thread` to exit
At the moment I guess it will exit when the replica shuts down, but for a clean shutdown it would be better to signal it.